src/service/engine/index.ts

```diff
@@ -172,30 +144,26 @@ export class EngineService {
     name: string,
     options: CreateEngineOptions = {}
   ): Promise<EngineModel> {
-    const accountVersion = (await this.context.connection.resolveAccountInfo())
-      .infraVersion;
-    this.validateCreateOptions(accountVersion, options);
-    this.setDefaultCreateOptions(accountVersion, options);
+    this.validateCreateOptions(options);
+    this.setDefaultCreateOptions(options);
 
     const { fail_if_exists, ...createOptions } = options;
 
     let query = `CREATE ENGINE ${
       fail_if_exists ? "" : "IF NOT EXISTS "
     } "${name}"`;
 
-    const queryParameters: (string | number)[] = [];
-    const createParameterNames =
-      accountVersion >= 2
-        ? this.CREATE_PARAMETER_NAMES_V2
-        : this.CREATE_PARAMETER_NAMES;
+    const queryParameters: (string | number | boolean)[] = [];
```

Comment:
just not sure why `boolean` has been added here. I believe that is correct, but it is not clear why it was not in previous implementation and exists here.

Reply:
This is probably a mistake. I think that boolean is first used by AUTO_START option, which was added recently. Yet, it's good we updated this annotation

Reply:
Yeah, this was confusing me as well. I think the type checker was only checking one condition here? In any case, this is corrected now.